Makefile

```diff
@@ -338,28 +320,51 @@ site-devel: ## Launch the website
 site-check: ## Test the site's links
 	# TODO: Clean up to use htmltest
 
-.PHONY: e2e
-e2e: ## Run E2E tests against a real k8s cluster
+
+# Tools for testing and troubleshooting
+
+.PHONY: setup-kind-cluster
+setup-kind-cluster: ## Make a kind cluster with standard ports forwarded
 	./test/scripts/make-kind-cluster.sh
+
+.PHONY: install-contour-working
+install-contour-working: | setup-kind-cluster ## Install the local working directory version of Contour into a kind cluster
 	./test/scripts/install-contour-working.sh
+
+.PHONY: install-contour-release 
+install-contour-release: | setup-kind-cluster ## Install the release version of Contour in CONTOUR_UPGRADE_FROM_VERSION, defaults to latest
+	./test/scripts/install-contour-release.sh $(CONTOUR_UPGRADE_FROM_VERSION)
+
+.PHONY: e2e
+e2e: | install-contour-working run-e2e cleanup-kind ## Run E2E tests against a real k8s cluster
+
+.PHONY: run-e2e
+run-e2e:
 	ginkgo -tags=e2e -mod=readonly -skipPackage=upgrade -keepGoing -randomizeSuites -randomizeAllSpecs -slowSpecThreshold=15 -r -v ./test/e2e
+
+.PHONY: cleanup-kind
+cleanup-kind:
 	./test/scripts/cleanup.sh
 
-.PHONY: upgrade
-upgrade: ## Run upgrade tests against a real k8s cluster
-	./test/scripts/make-kind-cluster.sh
-	./test/scripts/install-contour-release.sh $(CONTOUR_UPGRADE_FROM_VERSION)
+.PHONY: load-contour-image-kind
+load-contour-image-kind:
```

Comment:
Can probably remove this since it's required to have the image exported first to disk at a specific place.

Reply:
Or if you want to keep, add a comment stating what needs set to properly use?

Reply:
I was just moving things around, but I'll have a look and see if we can use Make properly here, and have a rule that will rebuild the contour image if stuff has changed, which can then be a prereq for this rule.

Reply:
Agreed that if I keep, I'll add a comment which tells you what this step expects.

Reply:
I've updated with an explanatory comment and a `make help` line.